src/fmu/dataio/dataio.py

```diff
@@ -469,20 +484,97 @@ def __post_init__(self) -> None:
         self._validate_content_key()
         self._validate_and_establish_fmucontext()
         self._validate_workflow_key()
-        self._update_globalconfig_from_settings()
 
         # check state of global config
         self._config_is_valid = global_configuration.is_valid(self.config)
         if self._config_is_valid:
             # TODO: This needs refinement: _config_is_valid should be removed
             self.config = global_configuration.roundtrip(self.config)
 
+        self._classification = self._get_classification()
+        self._rep_include = self._get_rep_include()
+
         self._establish_pwd_rootpath()
         logger.info("Ran __post_init__")
 
+    def _get_classification(self) -> enums.AccessLevel:
+        """
+        Get the security classification as an AccessLevel.
+        The order of how the classification is set is:
+        1. from classification argument if present
+        2. from access_ssdl argument (deprecated) if present
+        3. from access.classification in config (has been mirrored from
+        access.ssdl.access_level if not present)
+
```

Comment:
Maybe add
```
4. Default to internal if not found
```

...similar to docstring on _get_rep_include()

Not critical...

Reply:
yes, did not add it as you will never use that default.. that only happens if the config is invalid and then you won't create metadata anyway. I have a comment regarding this inside the code.
Will remove that default when we restructure the code to stop earlier and just do a pure object export if the config is invalid 🙂